pkg/solana/chainwriter/helpers.go

```diff
@@ -0,0 +1,218 @@
+package chainwriter
+
+import (
+	"context"
+	"crypto/sha256"
+	"errors"
+	"fmt"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/gagliardetto/solana-go"
+	"github.com/gagliardetto/solana-go/rpc"
+	"github.com/stretchr/testify/require"
+
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana/utils"
+)
+
+type TestArgs struct {
+	Inner []InnerArgs
+}
+
+type InnerArgs struct {
+	Address []byte
+}
+
+type DataAccount struct {
+	Discriminator        [8]byte
+	Version              uint8
+	Administrator        solana.PublicKey
+	PendingAdministrator solana.PublicKey
+	LookupTable          solana.PublicKey
+}
+
+// GetValuesAtLocation parses through nested types and arrays to find all locations of values
+func GetValuesAtLocation(args any, location string) ([][]byte, error) {
+	var vals [][]byte
+	path := strings.Split(location, ".")
+
+	addressList, err := traversePath(args, path)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, value := range addressList {
+		if byteArray, ok := value.([]byte); ok {
+			vals = append(vals, byteArray)
+		} else if address, ok := value.(solana.PublicKey); ok {
+			vals = append(vals, address.Bytes())
+		} else {
+			return nil, fmt.Errorf("invalid value format at path: %s", location)
+		}
+	}
+
+	return vals, nil
+}
+
+func GetDebugIDAtLocation(args any, location string) (string, error) {
+	debugIDList, err := GetValueAtLocation(args, location)
+	if err != nil {
+		return "", err
+	}
+
+	// there should only be one debug ID, others will be ignored.
+	debugID := string(debugIDList[0])
+
+	return debugID, nil
+}
+
+func GetValueAtLocation(args any, location string) ([][]byte, error) {
```

Comment:
Noticed this is essentially the same as `GetValuesAtLocation` and only really used in `GetDebugIDAtLocation` as far as I can tell. Couldn't we just use `GetValuesAtLocation` for that method as well?

Reply:
good call, fixed